src/python/pants/util/ordered_set.py

```diff
@@ -108,7 +70,7 @@ def __eq__(self, other: Any) -> bool:
         """Returns True if other is the same type with the same elements and same order."""
         if not isinstance(other, self.__class__):
             return NotImplemented
-        return self._items == other._items
+        return all(x == y for x, y in itertools.zip_longest(self._items, other._items))
```

Comment:
Is the idea here to early return? I believe `self._items == other._items` will already do this because it will resort to `dict'`s `__eq__` implementation, which is already highly optimized.

NB that this `__eq__` implementation is stricter than Twitter Commons. We enforce that it's the same class, e.g. `OrderedSet != FrozenOrderedSet`. This is to ensure that we do sane things with the engine and to mirror how in the stdlib `tuple != list`.

Reply:
The `__eq__` impl of the underlying (which was used before this edit) was not order sensitive for `dict`. And yea, the idea is to early return if possible.

Reply:
Ah, that makes a lot of sense. Thanks for the explanation.